Section/Clustering Page Rank (#2)
MrLongbottom/P9-Rapport: sections/method/Clustering-PageRank.tex

\section{Clustering Random Walk}

> To perform Random Walk on a document set, such as Nordjyske, we need a similarity measure between each document.

Author: kan du komme med en omformulering?

> To perform Random Walk on a document set, such as Nordjyske, we need a similarity measure between each document.

Reviewer: Omformulering i: https://github.com/MrLongbottom/P9-Rapport/pull/2/files#r504616547

> To perform random walk on a document set, such as Nordjyske, we need a similarity measure between each document.

Reviewer: ```suggestion
To perform random walk on a document set, such as Nordjyske, probabilities of jumping between documents is needed to perform each step.
We have chosen to use a similarity measure between each document as these probabilities.
```

These distributions can be arranged in a matrix $\theta$ which is described later.

In \cite{ClusterPageRank}, they describe the \gls{Cluster-CMRW} which is new version of the Random Walk model. 
They improve upon this model by incorporating information from clusters. 
The clusters are found by employing three different clustering algorithms.
They create a new transition probability matrix where the cluster information is added by combining three similarity functions based on the information levels present in the data.
They describe three similarity levels between:
\begin{itemize}
    \item Sentence to Sentence - $f(v_i \rightarrow v_j)$
    \item Sentence to Topic Cluster - $\omega(v_i, clus(v_i))$
    \item Topic Cluster to Document Set - $\pi(clus(v_i))$
\end{itemize}
where $v_i$ is the given sentence and $clus(v_i)$ is the cluster that $v_i$ belongs to.

We want to use this method to incorporate the information given by the topic distributions within a \gls{Cluster-CMRW}. 
We first create three similarity functions to incorporate topic cluster level information into a new adjacency matrix.
\begin{itemize}

>     \item Document to Document - $d2d(d_1, d_2)$

Reviewer: Skal der være en `\rightarrow` her ligesom i sentence to sentence?

Author: ```suggestion
    \item Document to Document - $d2d(d_1 $\rightarrow$ d_2)$
```

    \item Document to Topic Cluster - $d2t(d,t)$

>     \item Topic Cluster to Document Set - $t2D(t)$

Reviewer: Vi bruger ikke samme format som i den tidligere itemize. Er der en grund til det?

Author: Det er for at sige at det er anderledes end hvad det er i den originale artikel

Reviewer: Ok, men bliver man ikke forvirret når man kommer ned til den store similarity funktion (linje 36+), når den bruger samme format som artiklen, med f.eks. `clus(d_1)`, som her er `t`?

\end{itemize}

\noindent

> Where $d$ is a document, $t$ is a topic and $D$ is the whole document set.

Reviewer: D bliver ikke brugt i funktionerne, men er tilsyneladende bare en del af funktionsnavnet.

Author: ```suggestion
Where $d$ is a document, $t$ is a topic.
```

Notice that a document $d$ can contain multiple topics, as opposed to normal clustering where each element is generally assumed to only be part of one cluster.
These similarity functions are described in detail in \autoref{sec:similarity}.

To create the adjacency matrix, \cite{ClusterPageRank} linearly combine the different similarity functions, in which we intend to do the same.
Formally, the adjacency matrix is calculated with the following function.
$$ f(d_1 \rightarrow d_2 | clus(d_1), clus(d_2)) $$
which is evaluated to 

\begin{align*}
&f(d_1 \rightarrow d_2 | clus(d_1), clus(d_2)) = \\
&d2d(d_1, d_2) \cdot (\lambda \cdot t2D(clus(d_1))) \cdot d2t(d_1, clus(d_1)) \\ 
&+ (1-\lambda) \cdot t2D(clus(d_2)) \cdot d2t(d_2, clus(d_2))
\end{align*}
where $\lambda$ is a weight between $[0,1]$ that controls the relative contribution of the two clusters.

\section{Similarity Measures}\label{sec:similarity}
\subsection{Similarity between two documents}

> The matrix $\theta$ describes the topic distribution for each document $d \in M$, where $\theta_d$ is the topic distribution for document $d$, and is a $K$-dimensional vector that sums to 1, where $K$ is the number of topics.

Reviewer: Første gang M er nævnt, det er ikke tydeligt hvad den er.

Author: Det vil blive nævnt tidligere i LDA afsnittet

Reviewer: ```suggestion
The matrix $\theta$ describes the topic distribution for each document $d \in M$, where $\theta_d$ is the topic distribution for document $d$, and is a $K$-dimensional vector that sums to 1, where $K$ is the number of topics. \todo[inline]{Tjek at M er defineret når LDA afsnittet merges.}
```

> The similarity between two documents $d_1$ and $d_2$ as:

Reviewer: ```suggestion
The similarity between two documents $d_1$ and $d_2$ is defined as:
```

$$ d2d(d_1, d_2) = \sum_{1}^{K} min(\theta_{d_1,k}, \theta_{d_2,k})$$

> \subsection{Similarity between documents and topics}

Reviewer: ```suggestion
\subsection{Similarity between documents and topic clusters}
```

> We define the similarity between documents as the document-topic probability distribution between them.

Reviewer: ```suggestion
We define the similarity between a document and a topic cluster as the document-topic probability distribution between them.
```

> We define the similarity between documents as the document-topic probability distribution between them.

Reviewer: Der springes vidst mellem at vi kalder dem topics og topic clusters igennem afsnittet. Det er måske værd at overveje at kalde det én ting så meget som muligt.

Author: Yubs det er jeg enig i.

Reviewer: Fikser vi det på den her PR, eller venter vi til det er merget?

$$ d2t(d,t) = \theta_{d,t}$$

> \subsection{Similarity measure between topic clusters and whole document set}

Reviewer: ```suggestion
\subsection{Similarity measure between topic clusters and the whole document set}
```

They improve upon this model by incorporating information from clusters. 
The clusters are found by employing three different clustering algorithms.
They create a new transition probability matrix where the cluster information is added by combining three similarity functions based on the information levels present in the data.
They describe three similarity levels between:
\begin{itemize}
    \item Sentence to Sentence - $f(v_i \rightarrow v_j)$
    \item Sentence to Topic Cluster - $\omega(v_i, clus(v_i))$
    \item Topic Cluster to Document Set - $\pi(clus(v_i))$
\end{itemize}
where $v_i$ is the given sentence and $clus(v_i)$ is the cluster that $v_i$ belongs to.

We want to use this method to incorporate the information given by the topic distributions within a \gls{Cluster-CMRW}. 
We first create three similarity functions to incorporate topic cluster level information into a new adjacency matrix.
\begin{itemize}
    \item Document to Document - $d2d(d_1 $\rightarrow$ d_2)$
    \item Document to Topic Cluster - $d2t(d,t)$
    \item Topic Cluster to Document Set - $t2D(t)$

> \end{itemize}

Reviewer: ```suggestion
\end{itemize}
\todo[inline]{Overvej at være konsistente i brugen af t/topics og clusters, for at forbedre forståeligheden af den længere ligning på 37+ pt.. Hvordan det præcis skal gøres er ikke bestemt. Det kan muligvis indebære at der gøres så der ikke springes mellem topics og topic clusters igennem afsnittet.}
```
